src/main/java/io/confluent/copycat/hdfs/HdfsSinkTask.java

```diff
@@ -43,37 +43,51 @@ public HdfsSinkTask() {
   public void start(Properties props) {
     try {
       HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
-      hdfsWriter = new HdfsWriter(connectorConfig);
-      context.resetOffset(hdfsWriter.getPreviousOffsets());
+      assignment = context.assignment();
+      hdfsWriter = new HdfsWriter(connectorConfig, assignment);
+      recover(assignment);
+      resetOffsets();
     } catch (ConfigException e) {
-      throw new CopycatRuntimeException(
+      throw new CopycatException(
           "Couldn't start HdfsSinkConnector due to configuration error.", e);
     }
+
   }
 
   @Override
   public void stop() throws CopycatException {
-    try {
-      hdfsWriter.close();
-    } catch (IOException e) {
-      throw new CopycatRuntimeException("Error closing the HDFS writer", e);
-    }
+    hdfsWriter.close();
+
   }
 
   @Override
   public void put(Collection<SinkRecord> records) throws CopycatException {
     String topic;
     int partition;
-    for (SinkRecord record: records) {
-      topic = record.getTopic();
-      partition = record.getPartition();
+    for (SinkRecord record : records) {
+      topic = record.topic();
+      partition = record.kafkaPartition();
       TopicPartition topicPart = new TopicPartition(topic, partition);
-      hdfsWriter.writeRecord(topicPart, record);
+      if (assignment.contains(topicPart)) {
+        hdfsWriter.writeRecord(topicPart, record);
+      } else {
+        log.warn("Seen record {} from not assigned TopicPartition {}", record, topicPart);
```

Comment:
Modified to use `log.error()` and throw a `CopycatException` for now. We probably use the proper exception once we have that in copycat.